spec-main/extensions-spec.ts

```diff
@@ -650,4 +650,16 @@ describe('chrome extensions', () => {
       expect(textContent).to.equal('script loaded ok\n');
     });
   });
+
+  describe('manifest v3', () => {
+    it('registers background service worker', async () => {
+      const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+      const registrationPromise = new Promise<string>(resolve => {
+        customSession.serviceWorkers.once('registration-completed', (event, scope) => resolve(scope));
+      });
+      const extension = await customSession.loadExtension(path.join(fixtures, 'extensions', 'mv3-service-worker'));
+      const swScope = await registrationPromise;
+      expect(swScope).equals(extension.url);
```

Comment:
I'd love to have a bit more of a detailed test, e.g can we test that the service worker code actually runs?

Reply:
`registration-completed` actually doesn't emit if a script error occurs during evaluation of the service worker script.

#### Failed startup with `--enable-logging`
```
(node:3140) ExtensionLoadWarning: Warnings loading extension at D:\work\electron-fiddles\mv3\extension:
  Unrecognized manifest key 'action'.
  Unrecognized manifest key 'options_page'.
  Permission 'activeTab' is unknown or URL pattern is malformed.
  Permission 'scripting' is unknown or URL pattern is malformed.

[3140:0201/225800.367:INFO:CONSOLE(1)] "service worker installed", source: chrome-extension://hmjaablpdnebjcdkdhbilbcmpjnmgdon/background.js (1)
[3140:0201/225800.368:INFO:CONSOLE(5)] "Uncaught ReferenceError: welp is not defined", source: chrome-extension://hmjaablpdnebjcdkdhbilbcmpjnmgdon/background.js 
(5)
[3140:0201/225800.369:ERROR:extensions_browser_client.cc(57)] Extension Error:
  OTR:     false
  Level:   1
  Source:  manifest.json
  Message: Service worker registration failed
  ID:      hmjaablpdnebjcdkdhbilbcmpjnmgdon
  Type:    ManifestError
```

#### Successful startup with `--enable-logging`
```
(node:22272) ExtensionLoadWarning: Warnings loading extension at D:\work\electron-fiddles\mv3\extension:
  Unrecognized manifest key 'action'.
  Unrecognized manifest key 'options_page'.
  Permission 'activeTab' is unknown or URL pattern is malformed.
  Permission 'scripting' is unknown or URL pattern is malformed.

[22272:0201/230048.867:INFO:CONSOLE(1)] "service worker installed", source: chrome-extension://hmjaablpdnebjcdkdhbilbcmpjnmgdon/background.js (1)
***registration-completed chrome-extension://hmjaablpdnebjcdkdhbilbcmpjnmgdon/
```
I logged the result of `registration-completed` at the end

Reply:
ok... that seems incidental to me and not inherent, i.e. it could change in future, but perhaps I don't fully understand what "registration completed" means.

Reply:
Normally this event would fire after [`navigator.serviceWorker.register('/sw.js')`](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/register) successfully resolves. In the case of Chrome extensions, it's done in the background. I could probably link to that MDN doc in the 'registration-completed' description.